hail/src/main/scala/is/hail/expr/ir/streams/EmitStream.scala

```diff
@@ -1547,7 +1555,7 @@ object EmitStream {
                 val pulledRight = mb.genFieldThisRef[Boolean]("left_join_right_distinct_pulledRight]")
 
                 val producer = new StreamProducer {
-                  override val length: Option[EmitCodeBuilder => Code[Int]] = leftProducer.length
+                  override val length: Option[EmitCodeBuilder => Code[Int]] = None
```

Comment:
Was this just a bug before?

Reply:
Yeah, an inner join certainly doesn't need to have the same length as either input.

Reply:
Agreed